flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/FailoverRegion.java

```diff
@@ -206,13 +206,21 @@ private void restart(long globalModVersionOfFailover) {
 		try {
 			if (transitionState(JobStatus.CREATED, JobStatus.RUNNING)) {
 				// if we have checkpointed state, reload it into the executions
-				//TODO: checkpoint support restore part ExecutionVertex cp
-				/**
 				if (executionGraph.getCheckpointCoordinator() != null) {
+					// we restart the checkpoint scheduler for
+					// i) enable new checkpoint could be triggered without waiting for last checkpoint expired.
+					// ii) ensure the EXACTLY_ONCE semantics if needed.
+					if (executionGraph.getCheckpointCoordinator().isPeriodicCheckpointingConfigured()) {
+						executionGraph.getCheckpointCoordinator().stopCheckpointScheduler();
+					}
+
 					executionGraph.getCheckpointCoordinator().restoreLatestCheckpointedState(
-							connectedExecutionVertexes, false, false);
+							connectedExecutionVertexes, false);
```

Comment:
I have a general a high-level question about this whole approach, that I forgot to ask in my previous review: do we even need to introduce this additional restore method. It seems to me that all it does is optimizing the state assignment a little bit for regional failover (at slightly increased complexity for the full recovery cases).

Instead, couldn't we just simply use the existing `restoreLatestCheckpointedState` method, without providing any indexes and just call it from here with `executionGraph.allVertices()`. State assignment is a simple meta data operation and should run very fast in general. As this will only modify the `restoreState` variable, only the vertices that are actually restarted for the failed region see the effect. The remaining vertices are not restarted and do not care about the change. We might need to change `setInitialState` for this by removing the precondition that checks for `CREATED` or only assign to instances in this state.
If this is just an optimization attempt for this special case, this could reduce the amount of changes and potentiall bugs by a lot. What do you think? Did you find any other reason why this whole index handling is required?

Reply:
In my point of view, we should not depend on execution's current status to determine whether to assign state.
With the given indices, we could figure out the logical plan containing exact executions needed to be assigned. While try to assign state to all executions and use execution's status to determine whether assigning initial state would cause potential bugs indeed.
Moreover, current implementation has already use one base `restoreLatestCheckpointedState` method to handle both situations to reduce potential bugs happening. On the other hand, I think [FLINK-10713](https://issues.apache.org/jira/browse/FLINK-10713) could also reuse new `restoreLatestCheckpointedState(List<ExecutionVertex>, boolean)` method.
I'm not sure whether I have expressed my thoughts clearly to convince you, but please leave any thoughts or concerns if further discussion still needed.